arm-network/2016-09-01/swagger/network.json

```diff
@@ -7708,10 +7710,15 @@
           "description": "Gets or sets the reference of the PublicIP resource"
         },
         "provisioningState": {
+          "readOnly": true,
           "type": "string",
           "description": "Gets provisioning state of the PublicIP resource Updating/Deleting/Failed"
         }
       },
+      "required": [
+        "subnet",
+        "publicIPAddress"
+      ],
       "description": "Properties of VirtualNetworkGatewayIPConfiguration"
     },
     "VirtualNetworkGatewayIPConfiguration": {
```

Comment:
same reason.
"properties" property of "VirtualNetworkGatewayIPConfiguration" should be required.